src/auth_client_demo_assets/svelte/src/stores/auth.js

```diff
@@ -0,0 +1,131 @@
+import { AuthClient } from "@dfinity/auth-client";
+import { readable } from "svelte/store";
+import { createActor, canisterId } from "../../../../declarations/whoami";
+
+const defaultOptions = {
+  /**
+   *  @type {import("@dfinity/auth-client").AuthClientCreateOptions}
+   */
+  createOptions: {
+    idleOptions: {
+      // Set to true if you do not want idle functionality
+      disableIdle: true,
+    },
+  },
+  /**
+   * @type {import("@dfinity/auth-client").AuthClientLoginOptions}
+   */
+  loginOptions: {
+    identityProvider:
+      process.env.DFX_NETWORK === "ic"
+        ? "https://identity.ic0.app/#authorize"
+        : `http://localhost:4943?canisterId=${process.env.CANISTER_ID_INTERNET_IDENTITY}#authorize`,
+  },
+};
+
+/**
+ *
+ * @param {import("@dfinity/agent").Identity} identity
+ * @returns {import("@dfinity/agent").ActorSubclass<import("../../../../declarations/whoami/whoami.did")._SERVICE>}
+ */
+function actorFromIdentity(identity) {
+  return createActor(canisterId, {
+    agentOptions: {
+      host:
+        process.env.DFX_NETWORK === "ic"
+          ? "https://icp-api.io"
+          : "http://localhost:4943",
+      identity,
+    },
+  });
+}
+
+/**
+ * @typedef {Object} Auth
+ * @property {boolean} isAuthenticated
+ * @property {import("@dfinity/agent").Identity | null} identity
+ * @property {import("@dfinity/agent").ActorSubclass<import("../../../../declarations/whoami/whoami.did")._SERVICE> | null} whoamiActor
+ * @property {AuthClient | null} authClient
+ * @property {boolean} isReady
+ * @property {() => void} login
+ * @property {() => void} logout
+ * @property {() => void} init
+ *
+ */
+
+/**
+ * @type {Auth}
+ */
+const initialAuth = {
+  isAuthenticated: false,
+  isReady: false,
+  login: () => {},
+  logout: () => {},
+  init: () => {},
+  identity: null,
+  authClient: null,
+  whoamiActor: null,
```

Comment:
don't think the actor should be part of the authentication store, probably worth separating this layer